wagtail/admin/views/pages/listing.py

```diff
@@ -101,51 +101,81 @@ def get(self, request, parent_page_id=None):
             self.locale = None
             self.translations = []
 
+        # Search
+        self.query_string = None
+        if "q" in self.request.GET:
+            self.search_form = SearchForm(
+                self.request.GET, placeholder=_("Search pages…")
+            )
+            if self.search_form.is_valid():
+                self.query_string = self.search_form.cleaned_data["q"]
+        else:
+            self.search_form = SearchForm(placeholder=_("Search pages…"))
+
         return super().get(request)
 
     def get_ordering(self):
-        ordering = self.request.GET.get("ordering", "-latest_revision_created_at")
-        if ordering not in [
+        valid_orderings = [
             "title",
             "-title",
-            "content_type",
-            "-content_type",
             "live",
             "-live",
             "latest_revision_created_at",
             "-latest_revision_created_at",
-            "ord",
-        ]:
-            ordering = "-latest_revision_created_at"
+        ]
+
+        if self.query_string:
+            # default to ordering by relevance
+            default_ordering = None
+        else:
+            default_ordering = "-latest_revision_created_at"
+            # ordering by page order is only available when not searching
+            valid_orderings.append("ord")
+
+            # ordering by content type not currently available when searching, due to
+            # https://github.com/wagtail/wagtail/issues/6616
+            valid_orderings.append("content_type")
+            valid_orderings.append("-content_type")
```

Comment:
A minor thing, but if this is the case we probably want to unset `sort_key` for the Type column `if self.query_string` is truthy, so that the Type header is not clickable.

I managed to do this by changing `columns` into a `cached_property` (because the view extends `BaseListingView` which does not have `get_columns()` like `IndexView` does) and dynamically adding something like:

```py
sort_key="content_type" if not self.query_string else None,
```